src/client/auth/scram.rs

```diff
@@ -299,7 +299,7 @@ fn h_i<M: Mac + Sync>(str: &str, salt: &[u8], iterations: usize, output_size: us
 
 /// Parses a string slice of the form "<expected_key>=<body>" into "<body>", if possible.
 fn parse_kvp(str: &str, expected_key: char) -> Result<String> {
-    if str.chars().nth(0) != Some(expected_key) || str.chars().nth(1) != Some('=') {
+    if !str.starts_with(expected_key) || str.chars().nth(1) != Some('=') {
```

Comment:
I'm constantly amazed at how many specific lints like this exist. It even handles the fact that it's negated!